(RK-5) Add Basic User Scenario Integration Test

cowofevil opened the pull request:
Create an integration test for a basic r10k workflow. Segregate integration
tests from existing acceptance and spec tests. Created a folder layout for
integration tests. Created a pre-suite and library for future r10k integration
tests.

Commit:
(RK-5) Add Basic User Scenario Integration Test

Create an integration test for a basic r10k workflow. Segregate integration
tests from existing acceptance and spec tests. Created a folder layout for
integration tests. Created a pre-suite and library for future r10k integration
tests.

zreichert commented on integration/tests/user_scenario/basic.rb:
```diff
@@ -0,0 +1,50 @@
+
+agents.each do |agent|
+  step "Run Puppet Agent"
+  on(agent, puppet('agent', '--test'), :acceptable_exit_codes => 2) do |result|
```
You should expand your acceptable_exit_codes to include 0.

cowofevil replied:
@zreichert The Puppet run should always produce a notify message which will always register as a change so in this case if the Puppet run didn't return "2" we know that a failure occurred. Adding "0" wouldn't necessarily hurt anything, but would technically make the validation looser.

Commit:
(RK-5) Fix PR 278 Review Issues

The "integration/configs/pe/centos-7-64mda" config used the wrong "platform"
identifier and it has been fixed. Also there was unused variables in the
"integration/tests/user_scenario/basic.rb" file that have been removed.
